src/Microsoft.Data/DataFrame.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A DataFrame 
+    /// </summary>
+    public partial class DataFrame
+    {
+        internal DataFrameTable _table;
+        public DataFrame()
+        {
+            _table = new DataFrameTable();
+        }
+
+        public long RowCount => _table.RowCount;
+        public int ColumnCount => _table.ColumnCount;
+        public IList<string> Columns()
+        {
+            var ret = new List<string>();
+            for (int ii = 0; ii < ColumnCount; ii++)
+            {
+                ret.Add(_table.Column(ii).Name);
+            }
+            return ret;
+        }
+        public BaseDataFrameColumn Column(int index) => _table.Column(index);
+        public void InsertColumn(int columnIndex, BaseDataFrameColumn column) => _table.InsertColumn(columnIndex, column);
+        public void SetColumn(int columnIndex, BaseDataFrameColumn column) => _table.SetColumn(columnIndex, column);
+        public void RemoveColumn(int columnIndex) => _table.RemoveColumn(columnIndex);
+        public void RemoveColumn(string columnName) => _table.RemoveColumn(columnName);
+        
+        public object this[long rowIndex, int columnIndex]
+        {
+            get
+            {
+                return _table.Column(columnIndex)[rowIndex];
+            }
+            set
+            {
+                _table.Column(columnIndex)[rowIndex] = value;
+            }
+        }
+
+        #region Operators
+        public IList<object> this[long rowIndex]
+        {
+            get
+            {
+                return _table.GetRow(rowIndex);
+            }
+            //TODO?: set?
+        }
+
+        public object this[string columnName]
+        {
+            get
+            {
+                int columnIndex = _table.GetColumnIndex(columnName);
+                if (columnIndex == -1) throw new ArgumentException($"{columnName} does not exist");
```

Comment:
Should `GetColumnIndex` throw instead? Or is asking for an unknown column a reasonable thing to do?

Reply:
I'm not throwing yet because I want to support things like:

df["Int3"] = df["Int1"] + df["Int2"];

where df["Int3"] will be created if it doesn't exist. So, I'm not throwing yet. I don't like returning -1 either, so if I can write the setter (when I get around to it) without the -1, I'll add code to throw then.